// abstracts/db/index.js
var mongoClient = require('mongodb').MongoClient;
var singleDb;

var conSet = {
	host: 'localhost',
	port: '27017',
	db: 'abstracts'
}

exports.connect = function(callback) {
	if (singleDb === undefined) {
		mongoClient.connect('mongodb://' + conSet.host + ':' + conSet.port + '/' + conSet.db, function(err, db) {
			if (err) return callback(err);
			singleDb = db;
			callback(null, db);
		})
	} else {
		callback(null, singleDb);
	}
}


// AbstractProvider.prototype.getCollection = function(callback) {
// 	this.db.collection('abstracts', function(error, abstract_collection) {
// 		if (error) callback(error);
// 		else callback(null, abstract_collection);
// 	});
// };

// AbstractProvider.prototype.findAll = function(callback) {
// 	this.getCollection(function(error, abstract_collection) {
// 		if (error) callback(error);
// 		else {
// 			abstract_collection.find().toArray(function(error, results) {
// 				console.log(results);
// 				if (error) callback(error);
// 				else callback(null, results);
// 			});
// 		}
// 	});
// }

// AbstractProvider.prototype.save = function(abstracts, callback) {
// 	this.getCollection(function(error, abstract_collection) {
// 		if (error) callback(error);
// 		else {
// 			if (typeof(abstracts.length) == "undefined") {
// 				abstracts = [abstract];
// 			}

// 			for (var i = 0; i < abstracts.length; i++) {
// 				abstract = abstracts[i];
// 				abstract.created_at = new Date();
// 			}

// 			abstract_collection.insert(abstracts, function() {

// 			});
// 		}
// 	})
// }

// exports.AbstractProvider = AbstractProvider;
